crates/vim/src/motion.rs

```diff
@@ -1115,6 +1128,29 @@ fn window_bottom(
     }
 }
 
+fn previous_word_end(
```

Comment:
@ConradIrwin This is how I've implemented this feature right now but the `is_boundary` function is proving to be a bit limiting for certain cases (see failing test case). I think to deal with corner cases I might need to write completely custom logic and not use `find_preceding_boundary`. Thoughts?

Reply:
I am totally fine if a custom loop is easier to write! The main thing that `find_preceding_boundary` gives you is the utf8 offset calculation for multibyte chars, but that's easy to copy.

What is the edge-case you're struggling with?

(as always, happy to help if you want to book time: https://calendly.com/conradirwin/pairing)

Reply:
Looking more at the code, would it be better to do more in the `find_preceding_boundary` so we don't need to do the previous_word_start/next_word_end after it's done. It looks like we have the same bug in `previous_word_start` too, so fixing that might help.

From the vim help:

```
A word consists of a sequence of letters, digits and underscores, or a
sequence of other non-blank characters, separated with white space (spaces,
tabs, <EOL>).  This can be changed with the 'iskeyword' option.  An empty line
is also considered to be a word.
```

So it seems like at a first pass a word end is the preceding boundary where `left is alphanumeric and right is not, OR left is punctuation and right is alphanumeric, OR left and right are both newlines`?